scripts/analytics/trialClassAverageCompletion.js

```diff
@@ -0,0 +1,234 @@
+// Find average classroom completion rate for first week of cs1 activity
+
+// Algorithm
+// 1. Find initial classroom student activity for first cs1 level between start and midpoint times
+// 2. Find course instances for newly active students
+// 3. Find free teachers for student course instances
+// 4. Find first non-empty cs1 course instances for teachers
+// 5. Find per-classroom cs1 course completion averages
+// 6. Compute overall average
+
+// NOTE: update the endDay variable below to control the date range
+// NOTE: assumes a couple unchanged course and first level Ids
+
+// TODO: use lodash
+// TODO: better handling of archived classrooms
+// TODO: corner cases around classroom membership changes not reflected in course instances
+
+if (process.argv.length !== 4) {
+  console.log("Usage: node <script> <mongo connection Url> <mongo connection Url level session>");
+  console.log("Include ?readPreference=secondary in connection URLs");
+  process.exit();
+}
+
+'use strict';
+const scriptStartTime = new Date();
+const co = require('co');
+const MongoClient = require('mongodb').MongoClient;
+const mongoose = require('mongoose');
+const mongoConnUrl = process.argv[2];
+const mongoConnUrlLevelSessions = process.argv[3];
+
+const debugOutput = false;
+
+const cs1CampaignId = '55b29efd1cd6abe8ce07db0d';
+const cs1CourseId = '560f1a9f22961295f9427742';
+
+const daysToFirstLevelSession = 7;
+const daysToCompleteTrial = 7;
+const endDay = "2017-01-26";
+let startDay = new Date(`${endDay}T00:00:00.000Z`);
+startDay.setUTCDate(startDay.getUTCDate() - daysToCompleteTrial);
+const midpointDay = startDay.toISOString().substring(0, 10);
+startDay.setUTCDate(startDay.getUTCDate() - daysToFirstLevelSession);
+startDay = startDay.toISOString().substring(0, 10);
+const startObjectId = objectIdWithTimestamp(new Date(`${startDay}T00:00:00.000Z`));
+const midpointObjectId = objectIdWithTimestamp(new Date(`${midpointDay}T00:00:00.000Z`));
+const endObjectId = objectIdWithTimestamp(new Date(`${endDay}T00:00:00.000Z`));
+console.log(`Measuring days ${startDay} to ${midpointDay} to ${endDay}`);
```

Comment:
Script needs to error out if `new Date(`${endDay}T00:00:00.000Z`) > new Date()`. Time zones don't make it obvious when that's the case (it's at 4pm the day before)